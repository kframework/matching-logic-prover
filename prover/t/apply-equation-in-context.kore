symbol foo

axiom a: \exists{X{Int}} foo(\and(X{Int}, \equals(X{Int}, 3), 2, \equals(X{Int}, 2)))

claim \exists{X{Int}} foo(\and(X{Int}, \equals(X{Int}, 3), X{Int}, \equals(X{Int}, 2)))
strategy apply-equation(eq: \equals(X{Int}, #Y), idx: 1, direction: ->, at: 2)
       . apply(a, fail)
