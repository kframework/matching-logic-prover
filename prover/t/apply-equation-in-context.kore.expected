<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom a : \exists { X { Int } , .Patterns } foo ( \and ( X { Int } , \equals ( X { Int } , 3 ) , 2 , \equals ( X { Int } , 2 ) , .Patterns ) , .Patterns )
    </declaration> <declaration>
      axiom ax0 : \exists { X { Int } , .Patterns } foo ( \and ( X { Int } , \equals ( X { Int } , 3 ) , X { Int } , \equals ( X { Int } , 2 ) , .Patterns ) , .Patterns )
    </declaration> <declaration>
      symbol foo ( Int , Int , .Sorts ) : Int
    </declaration>
  </declarations>
</prover>
